src/user_input.rs

```diff
@@ -1,444 +1,814 @@
+                UserInput::Single(InputKind::SingleGamepadAxis(threshold)) => {
+                    if let Some(axes) = self.gamepad_axes {
+                        axes.get(GamepadAxis(gamepad, threshold.axis))
+                            .unwrap_or_default()
+                    } else {
+                        0.0
+                    }
+                }
+                UserInput::Single(InputKind::DualGamepadAxis(_)) => {
+                    if self.gamepad_axes.is_some() {
+                        self.get_input_axis_pair(input)
+                            .unwrap_or_default()
+                            .magnitude()
+                    } else {
+                        0.0
+                    }
+                }
+                UserInput::VirtualDPad { .. } => self
+                    .get_input_axis_pair(input)
+                    .unwrap_or_default()
+                    .magnitude(),
+                UserInput::Single(InputKind::GamepadButton(button_type)) => {
+                    if let Some(button_axes) = self.gamepad_button_axes {
+                        button_axes
+                            .get(GamepadButton(gamepad, *button_type))
+                            .unwrap_or_default()
+                    } else {
+                        0.0
+                    }
+                }
+                _ => use_button_value(),
+            }
+
+        // If there is no gamepad
+        } else {
+            match input {
+                UserInput::Single(
+                    InputKind::SingleGamepadAxis(_) | InputKind::GamepadButton(_),
+                ) => 0.0,
+                UserInput::Chord(chord) => {
+                    if let Some(input) = chord.get_at(0) {
+                        self.get_input_value(&UserInput::Single(*input))
+                    } else {
+                        0.0
+                    }
+                }
+                _ => use_button_value(),
+            }
+        }
+    }
+
+    /// Get the axis pair associated to the user input.
+    ///
+    /// See [`ActionState::action_axis_pair()`].
+    pub fn get_input_axis_pair(&self, input: &UserInput) -> Option<AxisPair> {
+        match input {
+            UserInput::Single(InputKind::DualGamepadAxis(threshold)) => {
+                if let Some(gamepad) = self.associated_gamepad {
+                    if let Some(axes) = self.gamepad_axes {
+                        let x = axes
+                            .get(GamepadAxis(gamepad, threshold.x_axis))
+                            .unwrap_or_default();
+                        let y = axes
+                            .get(GamepadAxis(gamepad, threshold.y_axis))
+                            .unwrap_or_default();
+                        Some(AxisPair::new(Vec2::new(x, y)))
+                    } else {
+                        Some(AxisPair::new(Vec2::ZERO))
+                    }
+                } else {
+                    Some(AxisPair::new(Vec2::ZERO))
+                }
```

Comment:
Providing fake axes values is confusing.

```suggestion
                    } else {
                        None
                    }
                } else {
                    None
                }
```